sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/_tests_/cas-invoices-batches-creation.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,284 @@
+import { INestApplication } from "@nestjs/common";
+import {
+  createE2EDataSources,
+  createFakeDisbursementValue,
+  E2EDataSources,
+  saveFakeApplicationDisbursements,
+  saveFakeCASSupplier,
+  saveFakeDisbursementReceiptsFromDisbursementSchedule,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { QueueNames } from "@sims/utilities";
+import { CASSupplierIntegrationScheduler } from "../cas-supplier-integration.scheduler";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+  mockBullJob,
+} from "../../../../../test/helpers";
+import {
+  CASInvoiceBatchApprovalStatus,
+  CASInvoiceStatus,
+  DisbursementValueType,
+  SupplierStatus,
+} from "@sims/sims-db";
+import { SystemUsersService } from "@sims/services";
+import { CASInvoicesBatchesCreationScheduler } from "../cas-invoices-batches-creation.scheduler";
+
+const CAS_INVOICE_BATCH_SEQUENCE_NAME = "CAS_INVOICE_BATCH";
+const CAS_INVOICE_SEQUENCE_NAME = "CAS_INVOICE";
+
+describe(
+  describeProcessorRootTest(QueueNames.CASInvoicesBatchesCreation),
+  () => {
+    let app: INestApplication;
+    let processor: CASSupplierIntegrationScheduler;
+    let db: E2EDataSources;
+    let systemUsersService: SystemUsersService;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource } = await createTestingAppModule();
+      app = nestApplication;
+      systemUsersService = nestApplication.get(SystemUsersService);
+      db = createE2EDataSources(dataSource);
+      // Processor under test.
+      processor = app.get(CASInvoicesBatchesCreationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Update existing records to avoid conflicts between tests.
+      await db.disbursementReceiptValue.update(
+        {
+          grantType: "BCSG",
+        },
+        { grantAmount: 0 },
+      );
+      // Delete all existing invoices and related data.
+      await db.casInvoiceDetail.delete({});
+      await db.casInvoice.delete({});
+      await db.casInvoiceBatch.delete({});
+      // Reset sequence numbers.
+      await db.sequenceControl.delete({
+        sequenceName: CAS_INVOICE_BATCH_SEQUENCE_NAME,
+      });
+      await db.sequenceControl.delete({
+        sequenceName: CAS_INVOICE_SEQUENCE_NAME,
+      });
+    });
+
+    it("Should create a new CAS invoice batch for a full-time receipt when there is a e-Cert receipt with no invoice associated.", async () => {
```

Comment:
E2E looks great. Unless there are more E2E tests going to come here in upcoming PRs, I would ask to have an E2E with no pending invoices to process.

Reply:
There will be more E2Es. The next PR will be for UI, but UI and schedulers must receive more E2Es.